File: RUBY/PRACTICA1/Practica-1/irrgarten/shield.rb
# frozen_string_literal: true
module Irrgarten
  class Shield
    attr_reader :protection, :uses

    def initialize(protection,uses)
      @protection = protection.to_f
      @uses = uses.to_i
    end

    def protect
      if @uses > 0
        @uses -= 1
        @protection
      else
        0
      end
    end

    def to_s
      "W[#{@protection},#{@uses}]"
    end

    def discard
      Dice.discard_element(@uses)
    end
  end
end

File: JAVA/Irrgarten_JAVA/src/irrgarten/Shield.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten;

/**
 *
 * @author alejandro
 */
public class Shield {
    private float protection;
    private int uses;
        
    public Shield(float protection, int uses){
        this.protection=protection;
        this.uses=uses;
    }
        
    public float protect(){
        if(uses>0){
            uses--;
            return protection;
        }else{
            return 0;
        }
    }
        
    public String toString(){
        return "S[" + protection + "," + uses + "]";
    }
        
    public float getProtection(){
        return protection;
    }
        
    public int getUses(){
        return uses;
    }
        
    public boolean discard(){
        return Dice.discardElement(uses);
    }
}
